### PR TITLE
CDR-1487 Add workflow_run for robot PR reports

### PR DESCRIPTION
**Pull request by alexlehn**
# Changes

Adding github action workflow_run for robot report on PR. 

# Pre-Merge checklist

- [ ] New code is tested
- [ ] Present and new tests pass
- [ ] Documentation is updated
- [ ] The build is working without errors
- [ ] No new Sonar issues introduced
- [ ] Changelog is updated
- [ ] Code has been reviewed

### CODE REVIEW

File: .github/workflows/report-robot-results.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    name: #robot-Report
+    runs-on: ubuntu-latest
+    if: github.event.workflow_run.conclusion == 'success'
+    steps:
+
+      # Download jacoco overall coverage from build & test output
+      - name: Download - Robot Results
+        uses: actions/download-artifact@v4
+        with:
+          github-token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```
Comment (vidi42) [inline .github/workflows/report-robot-results.yml:34]: Should it maybe be the `BOT_ACCESS_TOKEN`?

File: .github/workflows/report-robot-results.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: "Collect Robot Results"
```
Comment (vidi42) [inline .github/workflows/report-robot-results.yml:1]: Do we need to remove `Download - Robot results` the step from the `Build & Test` worfklow

File: .github/workflows/report-robot-results.yml (hunk 1 above)
Comment (alexlehn) [reply .github/workflows/report-robot-results.yml]: oops ... fixed thx

File: .github/workflows/report-robot-results.yml (hunk 2 above)
Comment (alexlehn) [reply .github/workflows/report-robot-results.yml:1]: yes in a dedicated PR, I want to try it out before removing it - chicken eg problem, the new workflow_run can only be used when it is available in the default branch => `develop`.

**Review by vidi42 — APPROVED** [review]

**Pull request merged**